### PR TITLE
Git fetch+reset instead of clean+fetch+merge

### PR DESCRIPTION
**Pull request by Alkarex**
Fix https://github.com/FreshRSS/FreshRSS/issues/2619
Avoid potentially dangerous git clean, and use more robust fetch + reset strategy instead

### CODE REVIEW

**Comment by Alkarex** [comment]
Just tested with various conflicts and seems to work fine

File: app/Controllers/updateController.php [hunk 1]
```diff
@@ -29,20 +29,22 @@ public static function hasGitUpdate() {
...
-			Minz_Log::warning('git pull error:' . $e->getMessage());
+			Minz_Log::warning('Git error:' . $e->getMessage());
+			if ($output == '') {
+				$output = $e->getMessage();
+			}
+			$return = 1;
 		}
 		chdir($cwd);
+		deleteInstall();
 		$line = is_array($output) ? implode('; ', $output) : '' . $output;
```
Comment (marienfressinaud) [inline app/Controllers/updateController.php:48]: `$output` is no longer an array here if I'm not wrong

**Review by marienfressinaud — APPROVED** [review]
Besides my comment, it looks good to me :+1:

File: app/Controllers/updateController.php (hunk 1 above)
Comment (Alkarex) [reply app/Controllers/updateController.php:48]: It risks being an array from `exec()`

**Pull request merged**